### PR TITLE
Wire authority override coming from discovery

### PR DESCRIPTION
**Pull request by zaharidichev**
This PR uses the latest changes to the proxy-api and consumes the authority override data, adding to the `Metadata` struct

Signed-off-by: Zahari Dichev <zaharidichev@gmail.com>

### CODE REVIEW

File: linkerd/app/Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ http = "0.1"
 http-body = "0.1"
 hyper = "0.12"
 linkerd2-metrics = { path = "../metrics", features = ["test_util"] }
-linkerd2-proxy-api = { git = "https://github.com/linkerd/linkerd2-proxy-api", features = ["arbitrary"], tag = "v0.1.11" }
```
Comment (kleimkuhler) [inline linkerd/app/Cargo.toml]: Do we want to block this getting a linkerd2-proxy-api release out first?

Comment (zaharidichev) [reply linkerd/app/Cargo.toml]: Yes we do

File: linkerd/proxy/api-resolve/src/pb.rs [hunk 2]
```diff
@@ -56,6 +60,19 @@ fn to_id(pb: TlsIdentity) -> Option<identity::Name> {
...
 }
 
+pub(in crate) fn to_authority(o: AuthorityOverride) -> Option<Authority> {
+    match o.authority_override.parse() {
+        Ok(name) => Some(name),
+        Err(_) => {
+            tracing::warn!(
+                "Ignoring invalid authority override: {}",
+                o.authority_override
+            );
```
Comment (olix0r) [inline linkerd/proxy/api-resolve/src/pb.rs]: Do we `warn` elsewhere in this file? Generally, if there's nothing a user can do about it, I prefer to keep this at debug. IMO, if the destination service is providing invalid values, that's an internal error and not a user error.

File: linkerd/app/Cargo.toml (hunk 1 above)
Comment (zaharidichev) [reply linkerd/app/Cargo.toml]: @kleimkuhler We are using the released api now.

**Pull request merged**